datahub-web-react/src/app/shared/recommendation.tsx

```diff
@@ -0,0 +1,44 @@
+import { useGetSearchResultsForMultipleQuery } from '../../graphql/search.generated';
+import { EntityType } from '../../types.generated';
+
+export function useGetRecommendedTags() {
+    const { data } = useGetSearchResultsForMultipleQuery({
+        variables: {
+            input: {
+                types: [EntityType.Tag],
+                query: '*',
+                start: 0,
+                count: 5,
+            },
+        },
+    });
+    return data?.searchAcrossEntities?.searchResults;
+}
+
+export function useGetRecommendedOwners() {
+    const { data } = useGetSearchResultsForMultipleQuery({
+        variables: {
+            input: {
+                types: [EntityType.CorpGroup, EntityType.CorpUser],
+                query: '*',
+                start: 0,
+                count: 5,
+            },
+        },
+    });
+    return data?.searchAcrossEntities?.searchResults;
+}
+
+export function useGetRecommendedDomains() {
+    const { data } = useGetSearchResultsForMultipleQuery({
+        variables: {
+            input: {
+                types: [EntityType.Domain],
+                query: '*',
+                start: 0,
+                count: 5,
+            },
+        },
+    });
+    return data?.searchAcrossEntities?.searchResults;
+}
```

Comment:
while I don't feel too strongly, I think it might make sense to just have one custom hook here that takes in a list of `entityTypes` that you pass into the query here. That way we only need one custom hook and for future use cases we don't have to copy paste and hardcode another one with different entity type(s)

Reply:
Created a custom hook :

`import { useGetSearchResultsForMultipleQuery } from '../../graphql/search.generated';
import { EntityType } from '../../types.generated';

export const useGetRecommendations = (types: Array<EntityType>) => {
    const { data } = useGetSearchResultsForMultipleQuery({
        variables: {
            input: {
                types,
                query: '*',
                start: 0,
                count: 5,
            },
        },
    });

    const recommendedData = data?.searchAcrossEntities?.searchResults?.map((searchResult) => searchResult.entity) || [];
    return [recommendedData];
};
`
And Use this like below.
`    const [recommendedData] = useGetRecommendations([EntityType.Domain]);
`